css/selectors/focus-visible-007.html

```diff
@@ -10,98 +10,47 @@
+        outline: red solid 5px;
+      }
     }
 
-    [data-hadmousedown] :focus-visible {
-      outline: red solid 5px;
+    :focus-visible {
+      outline: green solid 5px;
     }
 
-    [data-hadkeydown] :focus:not(:focus-visible) {
-      outline: 0;
-      background-color: tomato;
+    :focus:not(:focus-visible) {
+      outline: blue solid 5px;
     }
-
-    [data-hadmousedown] :focus:not(:focus-visible) {
-      outline: 0;
-      background-color: darkseagreen;
-    }
-
   </style>
 </head>
 <body>
   This test checks that using the keyboard in a way that does not move focus still causes <code>:focus-visible</code> matching to trigger.
   <ol id="instructions">
     <li>If the user-agent does not claim to support the <code>:focus-visible</code> pseudo-class then SKIP this test.</li>
-    <li>Use the mouse to focus the element below that says "Click me first."</li>
-    <li>If the element has a red outline, then the test result is FAILURE.</li>
+    <li>Use the mouse to focus the element below that says "Click me."</li>
+    <li>If the element has a blue outline, then the test result is a SUCCESS.</li>
     <li>Press the SHIFT key.</li>
-    <li>If the element now has a red background, then the test result is FAILURE.</li>
-    <li>Use the mouse to click the element below that says "Click me second."</li>
-    <li>If the element has a green background, the test result is SUCCESS.</li>
+    <li>If the element now has green outline, then the test result is SUCCESS.</li>
   </ol>
 
-  <div id="one" tabindex="0">Click me first.</div>
-  <div id="two" tabindex="0">Click me second.</div>
+  <div id="target" tabindex="0">Click me.</div>
   <script>
-    document.body.addEventListener("keydown", (e) => {
-      delete document.body.dataset.hadmousedown;
-      document.body.dataset.hadkeydown = "";
-    }, true);
-
-    document.body.addEventListener("mousedown", (e) => {
-      delete document.body.dataset.hadkeydown;
-      document.body.dataset.hadmousedown = "";
-    }, true);
-
     async_test(function(t) {
-      let tested_modality_change = false;
-      let tested_modality_unchanged_by_mouse_click = false;
-      let tested_mouse_focus_after_modality_change = false;
-
-      // TODO(crbug.com/828858): Remove this check once bug is resolved.
-      test_driver.send_keys(document.body, "\uE050").then(t.step_func(() => {
-        const handle_initial_focus = t.step_func(() => {
-          one.addEventListener("keyup", t.step_func(test_modality_change));
-          one.removeEventListener("focus", handle_initial_focus);
+      target.addEventListener("focus", t.step_func(function() {
+        assert_equals(getComputedStyle(target).outlineColor, "rgb(0, 0, 255)", `outlineColor for ${target.tagName}#${target.id} should be blue`);
 
-          test_driver.send_keys(one, "\uE050");
-        });
-
-        const test_modality_change = t.step_func(() => {
-          assert_equals(getComputedStyle(one).outlineColor, "rgb(59, 153, 252)");
```

Comment:
These asserts that were checking "weird" colors are the wrong assertions.

Reply:
That's odd, they would have passed when I added them. When did they begin failing?

Reply:
It looks like this is somehow failing since long time ago, when `test_driver.send_keys()` was implemented in Chromium: https://chromium-review.googlesource.com/c/chromium/src/+/1652635

Reply:
Looking at https://crbug.com/976438, I see the original version of this test is also triggering a leak. I'm not sure whether that's a problem with the test, or the code under test.

Reply:
That leak is due to `test_driver.click()` that is causing a leak on Chromium unless we wait for the promise to resolve.

The Chromium bug is https://bugs.chromium.org/p/chromium/issues/detail?id=1164600

I've fixed similar leaks in focus-visible tests in https://bugs.chromium.org/p/chromium/issues/detail?id=1163172

And once the fixed version of `focus-visible-007` gets imported into Chromium, I'll check if it has leaks or not there, and fix them in a similar way.

Reply:
Wonderful, thanks for the investigation and explanation!